coalib/settings/Setting.py

```diff
@@ -196,7 +197,7 @@ def __init__(self,
 
         self.from_cli = from_cli
         self.key = key
-        self.origin = str(origin)
+        self.origin = origin
```

Comment:
for backwards compatibility, we need `self.origin` to always return the filename.

So here, `self._origin = origin`

And then add

```py
@property
def origin(self):
    if isinstance(self._origin, SourcePosition):
        return self._origin.filename
    else:
        return self._origin
```

But please give it a good docstring to explain what it does.

Reply:
@jayvdb Although unrelated to this PR I don't quite understand why we have `self.key = key` here and then below also have
```
@property
def key(self):
```

Reply:
That looks like a bug to me.